internal/fuse/file.go

```diff
@@ -123,16 +125,15 @@ func (f *file) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadR
 	}
 
 	// Skip blobs before the offset
-	startContent := 0
-	for offset > int64(f.sizes[startContent]) {
-		offset -= int64(f.sizes[startContent])
-		startContent++
-	}
+	startContent := -1 + sort.Search(len(f.cumsize), func(i int) bool {
```

Comment:
My assumption would be that consecutive read calls usually access the same blob. In that case we could avoid the binary search altogether (starting from the second call) and cache the last accessed `startContent`.

Reply:
Perhaps, but I'm not convinced that this is the expensive part of the method. At 10TB and blocks of 1.5MB on average, binary search does 23 comparisons and incurs slightly fewer L1 cache misses. I can try to whip up a benchmark, but after I do the LRU thing.

Reply:
I wrote a benchmark and optimized the binary search, so it is now more than twice as fast. The benchmark suggests that reading a 200GiB file incurs about 1.5s of binary search overhead on my box. The actual figure might be a bit higher due to cache effects.

Caching the last startContent brings us back to the question of synchronization in Fuse, which I would rather avoid.